types/next/index.d.ts

```diff
@@ -11,7 +13,44 @@
 import * as http from "http";
 import * as url from "url";
 
+import { Response as NodeResponse } from "node-fetch";
+
 declare namespace next {
+    /**
+     * Context object used in methods like `getInitialProps()`
+     * <<https://github.com/zeit/next.js/issues/1651>>
+     */
+    interface NextContext {
+        /** path section of URL */
+        pathname: string;
+        /** query string section of URL parsed as an object */
+        query: {
+            [key: string]:
+                | boolean
+                | boolean[]
+                | number
+                | number[]
+                | string
+                | string[];
```

Comment:
I'll have to change this at the last minute to only `string` because calling this key through the `getInitialProps` of a stateless component trips up everything when passing it immediately as a prop, which will require unnecessary casting.